pandas/io/sql.py

```diff
@@ -114,6 +146,10 @@ def _parse_date_columns(data_frame, parse_dates):
             except TypeError:
                 fmt = None
             data_frame[col_name] = _handle_date_column(df_col, format=fmt)
+        # handle datetime Series with different offsets
+        # see GH30207
+        if _is_datetime_series_with_different_offsets(df_col):
+            data_frame[col_name] = to_datetime(df_col, utc=True)
```

Comment:
Just pass `utc=True` into `data_frame[col_name] = _handle_date_column(df_col, format=fmt)` above.

Reply:
Yes you are right it should be utc=True. I suppose this was already infered by pandas but we should always enforce it because as you mentionned:
> When reading TIMESTAMP WITH TIME ZONE types, pandas will convert the data to UTC. [source](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#datetime-data-types)

Please note that we still need the logic from L149 to L152 btw it should be "**elif** _is_datetime_series_with_different_offsets" (I shall make a commit soon with this correction and also the utc=True from above) in the case where we have different offsets (with postgres psycopg2 adds the offset to the local timezone which we do not want anyways) because is_datetime64tz_dtype(df_col) is False in this case.